bundles/org.openhab.core.ephemeris/src/main/java/org/eclipse/smarthome/core/ephemeris/EphemerisManager.java

```diff
@@ -27,57 +28,114 @@
 public interface EphemerisManager {
 
     /**
-     * Tests given day (related to today) status against configured weekend days
+     * Tests given day status against configured weekend days
      *
-     * @param offset Today +/- offset days (+1 = tomorrow, -1 = yesterday)
+     * @param date observed day
      * @return whether the day is on weekend
      */
-    boolean isWeekend(int offset);
+    boolean isWeekend(ZonedDateTime date);
 
     /**
-     * Tests given day (related to today) status against configured dayset
+     * Tests given day status against configured dayset
      *
      * @param daysetName name of the requested dayset, without prefix
-     * @param offset Today +/- offset days (+1 = tomorrow, -1 = yesterday)
-     * @return whether the day is on weekend
+     * @param date observed day
+     * @return whether the day is in the dayset
      */
-    boolean isInDayset(String daysetName, int offset);
+    boolean isInDayset(String daysetName, ZonedDateTime date);
 
     /**
-     * Tests given day status against official bank holidays
+     * Tests given day status
      *
-     * @param offset Today +/- offset days (+1 = tomorrow, -1 = yesterday)
+     * @param date observed day
      * @return whether the day is bank holiday or not
      */
-    boolean isBankHoliday(int offset);
+    boolean isBankHoliday(ZonedDateTime date);
 
     /**
      * Tests given day status against given userfile
      *
-     * @param offset Today +/- offset days (+1 = tomorrow, -1 = yesterday)
-     * @param filename absolute or relative path to the file on local file system
+     * @param date observed day
+     * @param source :
+     *            can be a String, absolute or relative path to the file on local file system
+     *            can be an URL, bundle resource file containing holiday definitions
      * @return whether the day is bank holiday or not
-     * @throws FileNotFoundException if given file does not exist
+     * @throws FileNotFoundException
      */
-    boolean isBankHoliday(int offset, String filename) throws FileNotFoundException;
+    boolean isBankHoliday(ZonedDateTime date, Object source) throws FileNotFoundException;
```

Comment:
I am not very happy with this API design. If we introduce a new parts for it then they should be clearly defined - my mean, your can vary.

Why not use two separate methods - one accepting a `String` parameter and the other one accepting an `URL` parameter.

Reply:
I'm not neither. The fact is that I introduced URL as a potential parameter for Unit testing (and maybe we could supply Holiday definitions in the bundle later. From scripting it will always be filename.

I did accept this **ugly** design because of the already long list of overloaded functions presented by the interface.

Do you still think we can keep it ?

Reply:
I look much better now. And the API is clear. I do not mind about one or two more methods in the interface.

If you do not want to make them public but only use them for unit testing then you do not need to define them in the interface.